### ISSUE TITLE
Installation of SWS arm64 on M2 macbook

### ISSUE THREAD

**Issue by dwadelson** [reaper-oss/sws#1757]
When installing reaper_sws-arm64.dylib via the .dmg download from https://www.sws-extension.org I got an instruction to copy the file into the UserPlugins Folder to install it, and the folder did not exist, to create one at /Library/Application Support/REAPER/UserPlugins

I created this folder and copied the file there, to no effect.

In fact, the file needed to be copied to /Applications/REAPER.app/Contents/Plugins, at least on my M2 macbook running Ventura 13.3.1

Took me much much much longer than it should have to figure it out, but I'm sure that other people even less capable than I am would appreciate having the instructions fixed

![image](https://user-images.githubusercontent.com/6504949/236043340-eb5f2539-1c10-49a0-888d-fc0a6a62b751.png)

**Comment by cfillion** [comment]
It needs to be copied to the UserPlugins directory in your REAPER install's resource path: Options > Show REAPER resource path in explorer/finder.

It's /Users/$USER/Library/Application Support/REAPER/UserPlugins for a regular non-portable install.

> In fact, the file needed to be copied to /Applications/REAPER.app/Contents/Plugins, at least on my M2 macbook running Ventura 13.3.1

That Plugins directory is reserved for REAPER's internal components.

**Comment by BPBaker** [comment]
I made a simple MacOS app in Automator to install the SWS .dylib to the default UserPlugins folder in the correct User Library, attached. The app also quarantines the .dylib so users don't have to "allow" it in MacOS System Preferences/Security. Would you consider adding this or a similar install application to the next build of SWS for MacOS?
[Install SWS Extension.zip](https://github.com/reaper-oss/sws/files/13182510/Install.SWS.Extension.zip)

**Comment by cfillion** [comment]
That app would need to itself be unquarantined...

**Comment by BPBaker** [comment]
True. I've instructed my students to command-click the app and select "Open" -- you still get the "this could be a malicious developer" popup, but at least if you "control-click Open" it gives you the option to open without having to go through the whole System Prefs/Security "allow" thing. It's a much smaller hurdle for people who are less tech savvy, in my experience. And unlike the outdated instructions on the .dmg, it gets the .dylib where it needs to go. But obviously the best thing would be have an Apple developer codesign... Is that possibly in the cards? 

(Ideally it'd also be nice to have an installer app with the same options as on Windows, where you can install to the default UserPlugins folder or to a portable install location. But my feeling is that portable install folks are already way more tech savvy, while the current MacOS install asks a lot of the average user.)

**Comment by cfillion** [comment]
I've been wondering whether a shell script that copies its arguments into the proper directory could solve this (or would that gets blocked by Apple too?) but haven't got around to testing it.

Something like "Drop files here.sh" instead of the current broken UserPlugins symlink:
```sh
# TODO: Detect what the file is and install into Scripts/UserPlugins/etc
# TODO: Handle dropping multiple files at once
# TODO: Check whether the resource path exists and report failure

cp "$1" ~/Library/'Application Support'/REAPER/UserPlugins/
```

If that's not possible (needs to be +x & not blocked), then the proper solution would be replacing the .dmg altogether with a .pkg installer. OSARA did that last time I checked.

**Comment by BPBaker** [comment]
I'll defer to your judgement on the best way to do it -- my application is just a stopgap to get a classroom of students up and running. 

It looks like OSARA uses a shell command but has users context menu-Open, too.

FWIW I did try doing this as shell script as a first-draft, actually (this was my first attempt at a shell script or using Automator). If fact, my app is just a wrapper for that script. The  script.command also had to be "allowed" in Sys. Prefs./Sec., but it didn't occur to me to try right-click/context "Open" until I had already made the Automator app. (And the app felt a bit cleaner for my purposes than the script.)
